### PR TITLE
ARCO-183: implement quarantine for problematic callback receivers

### PR DESCRIPTION
**Pull request by arkadiuszos4chain**
## Description of Changes

This update introduces a quarantine mechanism for receivers that fail to handle callbacks, helping to prevent repeated failed sends.

Changes:
* if a callback fails to send, the receiver is placed in quarantine
* during the quarantine period, all incoming callbacks for that receiver are persisted to storage
* the system periodically checks the stored callbacks and attempts to resend them once the quarantine period ends
* a cleanup process periodically prunes the store, removing outdated and unsent callbacks

## Testing Procedure

- [x] I have added new unit tests
- [x] All tests pass locally
- [x] I have tested manually in my local environment

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have updated `CHANGELOG.md` with my changes

### CODE REVIEW

**Review by pawellewandowski98 — APPROVED** [review]

File: internal/callbacker/store/postgresql/migrations/000002_add_quarantine_ts.up.sql [hunk 1]
```diff
@@ -0,0 +1,3 @@
+ALTER TABLE callbacker.callbacks ADD COLUMN quarantine_until TIMESTAMPTZ;
```
Comment (boecklim) [inline internal/callbacker/store/postgresql/migrations/000002_add_quarantine_ts.up.sql:1]: I think correct english would be `quarantined_until`, but not sure. 
Thinking of naming I think the callback itself is not quarantined, but rather the URL. 

What would you think of `postponed_until`?
IMO that would be a bit more expressive name of the field

File: internal/callbacker/send_manager.go [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+	}
+
+	m.run()
+	return m
+}
+
+func (m *sendManager) Add(entry *CallbackEntry) {
+	m.entriesWg.Add(1) // count the callbacks accepted for processing
+	go func() {
+		m.entries <- entry
```
[the remark this answers is not in the document]
Comment (arkadiuszos4chain) [reply internal/callbacker/send_manager.go:62]: It is called when the `entry` processing is complete in `run()` function

File: internal/callbacker/store/postgresql/migrations/000002_add_quarantine_ts.up.sql (hunk 1 above)
Comment (arkadiuszos4chain) [reply internal/callbacker/store/postgresql/migrations/000002_add_quarantine_ts.up.sql]: Agree, it's a better name.
Changed

File: internal/callbacker/quarantine_policy.go [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+type quarantinePolicy struct {
+	baseDuration time.Duration
+	abandonAfter time.Duration
+	now          func() time.Time
+}
+
+var infinity = time.Date(2999, time.January, 1, 0, 0, 0, 0, time.UTC)
+
+func (p *quarantinePolicy) Until(referenceTime time.Time) time.Time {
```
[the remark this answers is not in the document]
Comment (arkadiuszos4chain) [reply internal/callbacker/quarantine_policy.go:31]: I added a comment with description

**Review by boecklim — APPROVED** [review]

**Pull request merged**